dart/dynamics/ShapeFrame.cpp

```diff
@@ -57,7 +57,9 @@ DynamicsAspectProperties::DynamicsAspectProperties(
     const double frictionCoeff, const double restitutionCoeff)
   : mFrictionCoeff(frictionCoeff),
     mRestitutionCoeff(restitutionCoeff),
-    mSecondaryFrictionCoeff(frictionCoeff)
+    mSecondaryFrictionCoeff(frictionCoeff),
+    mFirstFrictionDirection(Eigen::Vector3d::Zero()),
```

Comment:
Maybe we want to set the default direction to non-zero direction (e.g., `Eigen::Vector3d::UnitZ()`) so that the default behavior is more consistent in terms of the friction direction?

Reply:
my intent was that an unset vector with zero length would use the global friction direction, and that only friction directions that are explicitly set to have a nonzero length would use their custom direction and frame. See the following logic in ContactConstraint.cpp

* https://github.com/dartsim/dart/blob/a932d87bbbf1a15fd83d2a2a8fb78fe9d4cb3bad/dart/constraint/ContactConstraint.cpp#L148-L172

does that make sense?

Reply:
Yeah, I read the logic but was suggesting changing the default behavior to using local friction direction (by setting `mFirstFrictionDirection` to non-zero vector intentionally) because first/second friction coefficients would make more sense when they're aligned with local directions (if they have different values).

Downside of it is that it changes the default simulation behavior (from using global friction direction to local friction direction) though. I'm okay to leave as it is as long as the logic is documented.

Reply:
a tricky aspect of the friction directions is choosing which to use when custom directions are set for both bodies. I've added logic in this PR to choose the direction from the body with the smaller primary friction coefficient, since that would be the dominant parameter, which makes sense to me, but I haven't tested it, and I'd be wary of making a big behavior change like that without testing.

Is the logic sufficiently documented? If not, I'll add the necessary documentation.

Reply:
Agreed to not changing the default behavior.

It's well documented in the implementation, but it'd be great if it's also documented in the header as well (maybe on the constructor?).

Reply:
how does cddef7a050b9aa9a5835d48a9b7c4f02de0f4934 look to you?

Reply:
Looks awesome! Thank you for adding it.